Automattic/wp-calypso
# Pro Dashboard: Add upgrade paths for downtime monitoring basic plan.

This PR is built on top of https://github.com/Automattic/wp-calypso/pull/79336

Related to 1204992567518369-as-1204995327932161

## Proposed Changes

This PR adds upgrade paths in the Notification Settings modal for the Downtime Monitoring Basic plan. The following components have been updated to show the **Upgrade** badge when hovered, will display the Upgrade popover.

* SMS Notification toggle
* Email Notification Add recipient button
* 1-minute interval option.

<img width="300" alt="Screen Shot 2023-07-14 at 7 33 37 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/4c318ded-b2ee-4438-a1b1-f28b7da931c8">  <img width="300" alt="Screen Shot 2023-07-14 at 7 52 48 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/eb95fcb1-b8c0-4319-8015-58d231b03f2c">


Additionally, For bulk update flow, at least one of the selected sites does not have a basic plan. We show a **Not available** badge instead of the **Upgrade** badge.

<img width="509" alt="Screen Shot 2023-07-25 at 2 40 54 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/1e984e5a-2408-4cd7-9d4e-8eb4d5ae8558">




_Note that the upgrade cost text will be added in a separate PR_
<img width="116" alt="Screen Shot 2023-07-14 at 7 55 57 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/1e77fe4c-30a1-463f-b310-1b1cae795787">


## Testing Instructions
Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

1. Check out this branch and run the local jetpack cloud server by executing `yarn start-jetpack-cloud`.
1. go to http://jetpack.cloud.localhost:3000/dashboard.
1. Select any site from the site table and click the monitor toggle to enable monitoring.
1. Click on the interval icon to show the Notification settings.
1. Confirm that the **Upgrade** badge is visible in the SMS Notification toggle, Add Email recipient button and 1-minute interval option. Make sure that when hovering the badge, the upgrade popover is displayed.

<img width="255" alt="Screen Shot 2023-07-14 at 7 43 24 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/dee38803-5a05-47b0-9e5f-c7e323418b0b">


### Bulk Update ###
For bulk updates. Confirm that the **Upgrade** badge is replaced with the **Disabled** badge.

### Temporarily Enable paid tier features. ###
* To test how the features would look when enabled. Modify the [following constant](https://github.com/Automattic/wp-calypso/pull/79420/files#diff-f35a450c1de6fd82f8c57a51c2645903df5dfee0a8b89847db7de1f960eaf200R107) to be **true**.

```
const hasDowntimeMonitoringPaidLicenses = true;
```

## Pre-merge Checklist
